src/Pages/ExplorePage/ExplorePage.jsx

```diff
@@ -0,0 +1,55 @@
+import { ExploreFilter } from "./ExploreFilter";
+import { useFilter, useFetch } from "../../CustomHooks/";
+import { useEffect } from "react";
+import { VideoCard } from "../../Components/VideoListing/";
+export const ExplorePage = () => {
+  const { filterState, filterDispatch } = useFilter();
+  const { data, loader, serverCall: fetchVideos } = useFetch();
+
+  useEffect(() => {
+    if (data !== null) {
+      filterDispatch({ type: "SET_ITEMS", payload: data.videos });
+    } else {
+      fetchVideos({ method: "GET", url: "/api/videos" });
+    }
+  }, [data]);
+
+  const filterByCategories = () => {
+    const { selectedCategory, items } = filterState;
+    if (selectedCategory === "All") {
+      return items;
+    }
+    return items.filter((item) => {
+      return item.category === selectedCategory;
+    });
+  };
+
+  const filteredData = filterByCategories();
+  console.log(filteredData);
+  const cards = filteredData.map((item) => {
+    return (
+      <VideoCard
+        key={item._id}
+        thumbnail={item.thumbnail}
+        title={item.title}
+        creator={item.creator}
+        views={item.views}
```

Comment:
you can pass the whole item as props and destructure there instead of passing one by one